js2-mode.el

```diff
@@ -12760,24 +12768,32 @@ i.e. ('name' 'value') = {name : { value: 3}}"
         (temp-object object)
         (temp t) ;temporay node
         (names prop-names))
-    (while (and temp names (js2-object-node-p temp-object))
+    (while (and temp names (or (js2-object-node-p temp-object)
+                               (js2-method-node-p temp-object)))
       (setq temp (js2-search-object temp-object (pop names)))
       (and (setq node temp)
-         (setq temp-object (js2-object-prop-node-right temp))))
+           (setq temp-object (js2-infix-node-right temp))))
     (unless names node)))
 
 (defun js2-search-scope (node names)
   "Searches NODE scope for jump location matching NAMES.
 NAMES is a list of property values to search for. For functions
 and variables NAMES will contain one element."
   (let (node-init
-        (val (js2-name-node-name (car names))))
-    (setq node-init (js2-get-symbol-declaration node val))
+        val
+        (first-name (car names)))
+    (cond
+     ((js2-name-node-p first-name)
+      (setq val (js2-name-node-name first-name))
+      (setq node-init (js2-get-symbol-declaration node val)))
+     ((js2-this-or-super-node-p first-name)
+      (setq node-init (js2-node-get-enclosing-scope-or-class (js2-node-get-enclosing-scope-or-class node))))
```

Comment:
This function is weird. You're only calling it in one place, twice. And it has two branches.

IIUC, the first time it will take one branch, and the other one the second time. Always in the same order. If so, this is unnecessarily ambiguous. Please rewrite it to only be called once and perform the whole computation inside. It will also need a different name; `js2-get-enclosing-class-or-object` is one option.